test_conformance/api/test_kernel_local_memory_size.cpp

```diff
@@ -0,0 +1,351 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include <cinttypes>
+
+#include "testBase.h"
+#include "harness/testHarness.h"
+#include <memory>
+
+static const char* empty_kernel = R"CLC(
+__kernel void empty_kernel() {
+}
+)CLC";
+
+static const char* local_memory_kernel = R"CLC(
+__kernel void local_memory_kernel(global int* data) {
+    __local int array[10];
+
+    size_t id = get_global_id(0);
+    array[id] = 2 * id;
+    data[id] = array[id];
+
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id == 0)
+    {
+        for(size_t i = 0; i < 10; i++)
+            data[id] += array[i];
+    }
+}
+)CLC";
+
+static const char* local_param_kernel = R"CLC(
+__kernel void local_param_kernel(__local int* local_ptr, __global int* src,
+    __global int* dst) {
+
+    size_t id = get_global_id(0);
+
+    local_ptr[id] = src[id];
+    barrier(CLK_GLOBAL_MEM_FENCE);
+    dst[id] = local_ptr[id];
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id == 9)
+    {
+        for(size_t i = 0; i < 10; i++)
+            dst[id] += local_ptr[i];
+    }
+}
+)CLC";
+
+static const char* local_param_local_memory_kernel = R"CLC(
+__kernel void local_param_local_memory_kernel(__local int* local_ptr,
+    __global int* src, __global int* dst) {
+
+    size_t id = get_global_id(0);
+
+    __local int local_data[10];
+    local_ptr[id] = src[id];
+
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id / 2 == 0) {
+        for(size_t i = 0; i < 10; i++)
+            local_data[id] += local_ptr[i];
+    }
+    else
+        local_data[id] = local_ptr[id] * 2;
+
+    barrier(CLK_LOCAL_MEM_FENCE);
+
+    dst[id] = local_data[id];
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(id == 9)
+    {
+        for(size_t i = 0; i < 10; i++)
+            dst[id] += local_data[i];
+        dst[id] += 666;
+    }
+}
+)CLC";
+
+int test_kernel_local_memory_size(cl_device_id deviceID, cl_context context,
+                                  cl_command_queue queue, int num_elements)
+{
+    int error;
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    // Check memory needed to execute empty kernel
+    if (create_single_kernel_helper(context, &program, &kernel, 1,
+                                    &empty_kernel, "empty_kernel")
+        != 0)
+    {
+        return TEST_FAIL;
+    }
+
+    cl_ulong kernel_local_usage = 0;
+    size_t param_value_size_ret = 0;
+    error = clGetKernelWorkGroupInfo(
+        kernel, deviceID, CL_KERNEL_LOCAL_MEM_SIZE, sizeof(kernel_local_usage),
+        &kernel_local_usage, &param_value_size_ret);
+    test_error(error,
+               "clGetKernelWorkGroupInfo for CL_KERNEL_LOCAL_MEM_SIZE failed");
+
+    test_assert_error(param_value_size_ret == sizeof(cl_ulong),
+                      "param_value_size_ret failed");
+
+    test_assert_error(kernel_local_usage >= 0, "kernel local mem size failed");
```

Comment:
I'm not going to lose sleep over this, but I don't think these "empty kernel" tests are adding anything, and I'd be fine removing them.

* The check for `param_value_size_ret` is covered by all of the other scenarios.
* Because `kernel_local_usage` is a `cl_ulong` the check for `kernel_local_usage >= 0` will always be true.